tests/assert/array_equals/udf.sql

```diff
@@ -1,4 +1,4 @@
-CREATE TEMP FUNCTION assert_array_equals(expected ANY TYPE, actual ANY TYPE) AS (
+CREATE OR REPLACE FUNCTION assert.array_equals(expected ANY TYPE, actual ANY TYPE) AS (
```

Comment:
What's the reasoning behind putting this in it's own dataset instead of leaving it as a temporary function? Is this a functional requirement for stored procedure tests?

Reply:
Good question. This came up because you cannot define a temporary udf and a permanent udf in the same query. So while we had to define the UDFs inside of the procedure as permanent, the assert udfs were all still temporary, causing it to fail. I figured it's easiest to rely on the existing infrastructure to transform the asserts to temporary only when they need to be.

Reply:
It's strange that you can't have temporary and permanent udfs in the same query. It seems reasonable to have these utilities in persistent locations, though.